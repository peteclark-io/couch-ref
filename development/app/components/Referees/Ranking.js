'use strict'

import React from 'react';
import classNames from 'classnames';

import styles from './styles.css';

const Ranking = React.createClass({

   propTypes: {
      rank: React.PropTypes.number,
      total: React.PropTypes.number,
      movement: React.PropTypes.number
   },

   suffix: (rank) => {
      if (rank % 10 === 1) {
         return 'st';
      } else if (rank % 10 === 2) {
         return 'nd';
      } else if (rank % 10 === 3) {
         return 'rd';
      }
      return 'th';
   },

   render: function() {
      return (
         <div className={classNames(styles['overall-ranking'], styles.centred)}>
            <h2 className={styles.rank}>Ranked {this.props.rank ? this.props.rank + this.suffix(this.props.rank) : 'Unranked'} <small>out of {this.props.total} Referees</small></h2>
         </div>
      );
   }
});

export default Ranking;
