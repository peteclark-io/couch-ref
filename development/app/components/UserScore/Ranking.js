'use strict'

import React from 'react';
import classNames from 'classnames';

import styles from './styles.css';

const Ranking = React.createClass({

   propTypes: {
      rank: React.PropTypes.number,
      movement: React.PropTypes.number
   },

   suffix: (rank) => {
      if (rank % 10 === 1) {
         return 'st';
      } else if (rank % 10 === 2) {
         return 'nd';
      } else if (rank % 10 === 3) {
         return 'rd';
      }
      return 'th';
   },

   render: function() {
      return (
         <div className={classNames(styles['overall-ranking'], styles.centred)}>
            <h1 className={styles.heading}>Your Ranking</h1>
            <h2 className={styles.rank}>{this.props.rank ? this.props.rank + this.suffix(this.props.rank) : 'Unranked'}!</h2>
         </div>
      );
   }
});

export default Ranking;
