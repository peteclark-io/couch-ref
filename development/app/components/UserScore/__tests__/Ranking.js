'use strict';

import React from 'react';
import {Ranking} from '../Ranking';
import renderer from 'react-test-renderer';
import {mount, shallow} from 'enzyme';

it('Renders score ok.', () => {
   const rendered = renderer.create(
      <Ranking rank={2} movement={1} />
   );

   expect(rendered.toJSON()).toMatchSnapshot();
});

it('Renders unrated if no ranking.', () => {
   const rendered = renderer.create(
      <Ranking />
   );

   expect(rendered.toJSON()).toMatchSnapshot();
});
