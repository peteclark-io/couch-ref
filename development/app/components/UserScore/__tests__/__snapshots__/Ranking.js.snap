exports[`test Renders score ok. 1`] = `
<div
  className="overall-ranking centred">
  <h1
    className="heading">
    Your Ranking!
  </h1>
  <h2
    className="rank">
    2nd
    !
  </h2>
</div>
`;

exports[`test Renders unrated if no ranking. 1`] = `
<div
  className="overall-ranking centred">
  <h1
    className="heading">
    Your Ranking!
  </h1>
  <h2
    className="rank">
    Unranked
    !
  </h2>
</div>
`;
